lib/product-import.coffee

```diff
@@ -133,6 +121,62 @@ class ProductImport
         Promise.resolve(@_summary)
     , {concurrency: 1} # run 1 batch at a time
 
+  _getExistingProductsForSkus: (skus) =>
+    new Promise (resolve, reject) =>
+      skuChunks = @_separateSkusChunksIntoSmallerChunks(skus, skus)
+      Promise.map(skuChunks, (skus) =>
+        new Promise (resolve, reject) =>
+          predicate = @_createProductFetchBySkuQueryPredicate(skus)
+          if Buffer.byteLength(encodeURIComponent(predicate),'utf-8') > 8072
+            errMessage = "product fetch query size: #{Buffer.byteLength(encodeURIComponent(predicate),'utf-8')} bytes, exceeded the supported " +
+            "size, please try with a smaller batch size."
+            @logger.error(errMessage)
+            reject (errMessage)
+          @client.productProjections
+          .where(predicate)
+          .staged(true)
+          .fetch()
+          .then ({ body: { results } }) ->
+            resolve(results)
+          .catch (err) -> reject(err)
+      , { concurrency: 5 })
+      .then (results) ->
+        debug 'Fetched products: %j', results
+        resolve(_.flatten(results))
+      .catch (err) -> reject(err)
+
+  ###*
+   * takes an array of sku chunks and returns an array of sku chunks
+   * where each chunk fits inside the query
+  ###
+  _separateSkusChunksIntoSmallerChunks: (skusChunk, skus) ->
+    # use two sku lists since we have to query for masterVariant and variant
+    # with the same list of skus
+    skuString = encodeURIComponent(
+      "\"#{skusChunk.join('","')}\"\"#{skusChunk.join('","')}\""
+    )
+    whereQuery = "
+      masterVariant(sku in ()) or variants(sku in ())
+    "
+    fixBytes = Buffer.byteLength(encodeURIComponent(whereQuery),'utf-8')
```

Comment:
Please consider here API domain which could be configured to some other internal URL and project key length also.